app/models/data_resources/resource_modules/media_content.rb

```diff
@@ -27,7 +27,19 @@ def convert_source_url_to_external_storage
     begin
       uri = URI.open(source_url.url)
       file = File.open(uri)
-      attachment.attach(io: file, filename: File.basename(file))
+
+      # Versuche den Dateinamen aus dem Header zu extrahieren
+      # bsp.: Content-Disposition: attachment; filename=quot.pdf;
+      content_disposition = uri.meta['content-disposition']
+      original_filename = content_disposition&.match(/filename="(.+)"/i)&.[](1)
```

Comment:
in welchen fällen hast du hier was gehabt?
ich habe eine bildurl genommen von unsplash: https://plus.unsplash.com/premium_photo-1666183671680-88209728a44f?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=2071&q=80

per byebug erhalte ich an der stelle

```
(byebug) uri
#<Tempfile:/var/folders/3f/73lvkym10b3dxhn_lf18c6280000gn/T/open-uri20230505-6951-1bq99ib>
(byebug) uri.meta['content-disposition']
nil
```